specs/utils.spec.js

```diff
@@ -639,46 +654,170 @@ describe( "utils", () => {
 
-			latest.mockImplementation( jest.fn( ( arg, cb ) => {
-				cb( null, "2.1.1" );
-			} ) );
+			return util.getAvailableVersionInfo().catch( err => {
+				expect( err ).toEqual( "nope" );
+			} );
+		} );
+	} );
+
+	describe( "detectVersion", () => {
+		describe( "when using a non-prerelease version", () => {
+			beforeEach( () => {
+				currentPackage.version = "4.5.0";
+				util.getAvailableVersionInfo = jest.fn( () => ( Promise.resolve( {
+					latestFullVersion: "4.5.0",
+					latestPrereleaseVersion: "5.0.0-refactor-and-jest.0"
+				} ) ) );
+			} );
+
+			describe( "and using the latest version", () => {
+				it( "should log the appropriate message", () => {
+					return util.detectVersion().then( () => {
+						expect( chalk.green ).toHaveBeenCalledTimes( 1 );
+						expect( chalk.green ).toHaveBeenCalledWith( "You're running the latest version (4.5.0) of tag-release." );
+					} );
+				} );
+			} );
+
+			describe( "and there is a newer version available", () => {
+				beforeEach( () => {
+					currentPackage.version = "4.4.0";
+				} );
+
+				it( "should log the appropriate message", () => {
+					return util.detectVersion().then( () => {
+						expect( chalk.red ).toHaveBeenCalledTimes( 2 );
+						expect( chalk.red.mock.calls[ 0 ] ).toEqual( [ "There is an updated version (4.5.0) of tag-release available." ] );
+						expect( chalk.red.mock.calls[ 1 ] ).toEqual( [ "To upgrade, run 'npm install -g tag-release'" ] );
+						expect( chalk.yellow ).toHaveBeenCalledTimes( 2 );
+						expect( chalk.yellow.mock.calls[ 0 ] ).toEqual( [ "4.5.0" ] );
+						expect( chalk.yellow.mock.calls[ 1 ] ).toEqual( [ "'npm install -g tag-release'" ] );
+					} );
+				} );
+			} );
+		} );
+
+		describe( "when using a prerelease version", () => {
+			beforeEach( () => {
+				util.getAvailableVersionInfo = jest.fn( () => ( Promise.resolve( {
+					latestFullVersion: "7.6.0",
+					latestPrereleaseVersion: "7.7.7-pre.0"
+				} ) ) );
+			} );
+			describe( "and it is the latest available prerelease version", () => {
+				beforeEach( () => {
+					currentPackage.version = "7.7.7-pre.0";
+				} );
 
-			return util.detectVersion().then( () => {
-				expect( chalk.red ).toHaveBeenCalledTimes( 2 );
-				expect( chalk.yellow ).toHaveBeenCalledTimes( 3 );
-				expect( chalk.red.mock.calls ).toEqual( [ [ "To upgrade run" ], [ "You're using an old version (1.0.0) of tag-release. Please upgrade to 2.1.1.\nTo upgrade run 'npm install tag-release -g'" ] ] );
-				expect( chalk.yellow.mock.calls ).toEqual( [ [ "1.0.0" ], [ "2.1.1" ], [ "'npm install tag-release -g'" ] ] );
+				it( "should log the appropriate message", () => {
+					return util.detectVersion().then( () => {
+						expect( chalk.green ).toHaveBeenCalledTimes( 1 );
+						expect( chalk.green ).toHaveBeenCalledWith( "You're running the latest pre-release version (7.7.7-pre.0) of tag-release." );
+					} );
+				} );
+			} );
+
+			describe( "and there is a newer prerelease version available", () => {
```

Comment:
It would be nice to see a test that examines the behavior when the current version is a prerelease, but there is a newer prerelease available for a different identifier... 

## Use Case

> current version 7.7.1-pre.0
> latestPrereleaseVersion 7.7.1-somethingelse.0

The above is per our Gryffindor Zoom discussion that if you are in prerelease land (current version is a prerelease) that you'd be notified of any other prerelease version (whether it is of the same identifier or another identifier)

**NOTE:** I'm not sure how `semver.gt( prv, memo )` will determine which prerelease version is greater if comparing `7.7.1-pre.0` with `7.7.1-somethingelse.0`. Based on the note above I was hoping we would pick whichever one was newer (which one was tagged latest), but comparing version numbers without date information probably won't give us the information we need. If this isn't easy to do, I'm open to redefining the behavior and just going with a simpler approach